app/controllers/amusement_parks_controller.rb

```diff
@@ -0,0 +1,13 @@
+class AmusementParksController < ApplicationController
+
+  def show
+    @park = AmusementPark.find(params[:id])
+    @mechanics = Mechanic.joins(:mechanic_rides, :rides).where("amusement_park_id = #{@park.id}").distinct
```

Comment:
There's an easier way to do this! 
1) Next time make sure to encapsulate this in a model method as it's more than just CRUD functionality

2) The easiest way to do this is to add an additional association in order to mooch off the rides associations since park has many rides! In the amusement_park model, you can add `has_many :mechanics, through: :rides`. This will allow you to then be able to reach a mechanic directly from a park: `park.mechanics` and so you can therefore do `park.mechanics.distinct`